src/CastleAdventure.java

```diff
@@ -3,29 +3,170 @@
-        Scanner scanner = new Scanner(System.in);
-
-        // Нижче перша частина завдання для прикладу
-        // Там де в завданні вказані початкові дані у вигляді масиву - ініціалізуйте окремий масив
-        // Нижче обробляйте масив, додайте логіку обробки
-        // Частина 1: Увійди в Замок
-        int[] secretCode = {7, 14, 28, 56, 112};
-        System.out.println("Введіть 5 чисел, щоб відкрити двері замку:");
-        boolean isCodeCorrect = true;
-
-        for (int i = 0; i < secretCode.length; i++) {
-            int userInput = scanner.nextInt(); // питаємо в користувача чергове число
-            if (userInput != secretCode[i]) { // перевіряємо число проти i-го елемента масиву з правильним кодом
-                isCodeCorrect = false; // якщо натрапляємо на неправильно введене число - усю комбінацію вважаємо невірною
+        System.out.println("Hello brave hero!");
+        System.out.println("Welcome into the 'Castle Adventure' journey!");
+        System.out.println("Your first task is to enter the castle. To open the door you need to enter correct password!");
+        System.out.println("Do you need a hint for password answer? Type 1 for 'yes' or 2 for 'no'");
+        Scanner denScanner = new Scanner(System.in);
+
+        
+        int choice1 = denScanner.nextInt();
+        if (choice1 == 1) {
+            System.out.println("1: How many oceans do we have?");
+            System.out.println("2: Favorite t-shirt number of Cristiano Ronaldo");
+            System.out.println("3: Devil's number");
+            System.out.println("4: 7*8-7");
+            System.out.println("5: 5\u00B3");
+        }
+        else if (choice1 == 2) {
+            System.out.println("It's your choice. Try to enter correct password");
+        } else {
+            System.out.println("Hmm.. Looks like you are not very attentive. Ok, never mind, try to enter correct password to enter the castle");
+        }
+
+        int[] secretCode = {5, 7, 13, 49, 125};
+        boolean isCodeCorrect = false;
+
+        while (!isCodeCorrect) {
+            System.out.println("Type 5 numbers to open the castle's door:");
+            isCodeCorrect = true;
+
+            for (int i = 0; i < secretCode.length; i++) {
+                int userInput = denScanner.nextInt();
+                if (userInput != secretCode[i]) {
+                    isCodeCorrect = false;
+                }
+            }
+
+            if (isCodeCorrect) {
+                System.out.println("Hooray! Door is open! Let's enter!");
+            } else {
+                System.out.println("Incorrect password");
+
+                System.out.println("\nWhat next?");
+                System.out.println("1 - Try again!");
+                System.out.println("2 - Get hint");
+                System.out.println("3 - Finish the game");
+
+                int choice2 = denScanner.nextInt();
+
+                if (choice2 == 2) {
+                    System.out.println("Hint:");
```

Comment:
You see here we can't avoid some code duplication right?

Now after we got new lessons you know that this could be extracted to some method to avoid code duplication, but here in this precise context - you can't avoid it :)

Just pointing how useful the methods are to put repeatable code in a single place